### ISSUE TITLE
Examples do not compile due to template deduction error (multi_fem)

### ISSUE THREAD

**Issue by jjwilke** [kokkos/kokkos#1928]
The multi_fem examples are broken

````
/Users/jjwilke/Programs/kokkos/cmake-1924/example/multi_fem/Nonlinear.hpp:349:9: 
error: no matching function for call to 'fill(size_t, int, Kokkos::CrsMatrix<double, Kokkos::OpenMP>::coefficients_type&)'
/Users/jjwilke/Programs/kokkos/cmake-1924/example/multi_fem/LinAlgBLAS.hpp:225:6: 
note:   template argument deduction/substitution failed:
````

**Comment by jjwilke** [comment]
The fill function assumes:

````
template< typename ScalarA ,
          typename ScalarX ,
          class L ,
          class D ,
          class MX >
void fill( const size_t n ,
           const ScalarA & alpha ,
           const View< ScalarX * , L , D , MX > & x )
{
  Impl::Fill< ScalarA , ScalarX , L , D >( n , alpha , x );
}
````
But the type of `x` passed in is sometimes
````
KokkosView<double,exec_space>
````
which only has 2 template parameters and does not specify layout, etc.

Functions should be updated to (or templated on the view type itself)
````
template< typename ScalarA ,
          typename ScalarX ,
          typename... Args
         >
void fill( const size_t n ,
           const ScalarA & alpha ,
           const View< ScalarX * , Args...> & x )
{
  Impl::Fill< ScalarA , ScalarX , Args...>( n , alpha , x );
}
````
to handle views with varying numbers of template parameters.

**Comment by jjwilke** [comment]
I can't self-assign, apparently? I think perhaps I am not in the kokkos group. This can get assigned to me.

**Comment by ndellingwood** [comment]
@ibaned I think you fixed this in PR #2181
